tracer/src/Datadog.Trace/Telemetry/TelemetrySettings.cs

```diff
@@ -0,0 +1,54 @@
+﻿// <copyright file="TelemetrySettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.Telemetry
+{
+    internal class TelemetrySettings
+    {
+        public TelemetrySettings(IConfigurationSource source, ImmutableTracerSettings tracerSettings)
+        {
+            TelemetryEnabled = source?.GetBool(ConfigurationKeys.Telemetry.Enabled) ??
+                               // default value
+                               true;
+
+            var requestedTelemetryUri = source?.GetString(ConfigurationKeys.Telemetry.Uri);
+
+            if (!string.IsNullOrEmpty(requestedTelemetryUri) && Uri.TryCreate(requestedTelemetryUri, UriKind.Absolute, out var telemetryUri))
+            {
+                // if we're using a custom telemetry Uri, then it could be an external intake and need the api key
+                TelemetryUrl = telemetryUri;
+                ApiKey = source.GetString(ConfigurationKeys.Environment);
```

Comment:
```suggestion
                ApiKey = source.GetString(ConfigurationKeys.ApiKey);
```

Reply:
😨 Definitely should have caught this in tests! Will commit suggestion, but will look into why it was missed too!